### PR TITLE
Filterbar on projects page doesn't work

### LINKED ISSUE #1319: Filterbar on projects page doesn't work
It changes url but doesn't actually filter results
Also there shouldn't be a project filter at all probably

### PR DESCRIPTION
**Pull request by sab94a**
## PR includes

- [x] Refactor

## Related issues

Resolve #1319

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/9020793/c92f6f65-7d25-4882-b20a-72708ac81e0d

### CODE REVIEW

File: src/components/ProjectListItem.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { ActivityByIdReturnType } from '../../trpc/inferredTypes';
 import { Table, TableCell, TableRow } from './Table';
 import { UserGroup } from './UserGroup';
 
-interface ProjectListItemProps {
+export interface ProjectListItemProps {
```
Comment (awinogradov) [inline src/components/ProjectListItem.tsx:10]: No way!

`React.ComponentProps`

File: src/components/CollapsableItem.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ const StyledCollapsableHeader = styled.div`
     padding-bottom: 1px;
 `;
 
-const StyledCollapsableItem = styled.div`
+export const StyledCollapsableItem = styled.div`
```
Comment (awinogradov) [inline src/components/CollapsableItem.tsx:39]: Sorry, but it makes code dirty. In time when you want to export part of component with styles you must create new component. Component style is private API.

File: src/components/CollapsableItem.tsx [hunk 3]
```diff
@@ -143,9 +144,10 @@ const StyledCollapsableContainer = styled.div<{ collapsed: boolean; deep: number
                 margin-left: -${collapseOffset}px;
             }
 
-            > ${StyledCollapsableItem}:before {
+            ${hasChild &&
+            `> ${StyledCollapsableItem}:not(:last-child):before {
```
Comment (awinogradov) [inline src/components/CollapsableItem.tsx:148]: I'm not sure anybody can understand what is this

File: src/components/ProjectListItemCollapsable/ProjectListItemCollapsable.tsx [hunk 4]
```diff
@@ -1,68 +1,55 @@
...
+    background-color: ${gray4};
+    border-radius: ${radiusM};
+    margin: 0px;
+    padding: 0px;
+`;
 
-const StyledNoGoals = styled(Text)`
-    white-space: nowrap;
+const StyledGoalsWrapper = styled(StyledCollapsableItem)`
+    padding-bottom: 1px;
```
Comment (awinogradov) [inline src/components/ProjectListItemCollapsable/ProjectListItemCollapsable.tsx:21]: Seriously?)

File: src/components/ProjectListItemConnected.tsx [hunk 5]
```diff
@@ -57,9 +54,9 @@ export const ProjectListItemConnected: FC<{
         setIsCollapsed((value) => !value);
     }, []);
 
-    const onGoalsClick = useCallback(() => {
-        setIsCollapsedGoals((value) => !value);
-    }, []);
+    if (!project._count.goals) {
```
Comment (awinogradov) [inline src/components/ProjectListItemConnected.tsx:57]: Add comment pls

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/ProjectListItem.tsx (hunk 1 above)
Comment (sab94a) [reply src/components/ProjectListItem.tsx]: fixed

File: src/components/CollapsableItem.tsx (hunk 2 above)
Comment (sab94a) [reply src/components/CollapsableItem.tsx]: fixed

File: src/components/CollapsableItem.tsx (hunk 3 above)
Comment (sab94a) [reply src/components/CollapsableItem.tsx]: add comment, will be refactored here https://github.com/taskany-inc/issues/issues/1448

File: src/components/ProjectListItemCollapsable/ProjectListItemCollapsable.tsx (hunk 4 above)
Comment (sab94a) [reply src/components/ProjectListItemCollapsable/ProjectListItemCollapsable.tsx]: fixed

File: src/components/ProjectListItemConnected.tsx (hunk 5 above)
Comment (sab94a) [reply src/components/ProjectListItemConnected.tsx:59]: fixed

**Comment by sab94a** [comment]
https://github.com/taskany-inc/issues/assets/9020793/098d0695-7d84-44be-a570-1106dc625e39

**Review by awinogradov — CHANGES_REQUESTED** [review]
image
Sorry, it's broken 😿

**Comment by sab94a** [comment]
https://github.com/taskany-inc/issues/assets/9020793/0cd8a862-c3c7-48e0-8117-db7500ef553b

File: src/components/ProjectsPage/ProjectsPage.tsx [hunk 6]
```diff
@@ -65,6 +67,8 @@ export const ProjectsPage = ({ user, ssrTime }: ExternalPageProps) => {
         goalsQuery: queryState,
     });
 
+    const tabsMenuOptions: Array<[string, string]> = [[tr('Goals'), routes.projects()]];
```
Comment (awinogradov) [inline src/components/ProjectsPage/ProjectsPage.tsx:70]: Types are unnecessary here

**Review by awinogradov — APPROVED** [review]

File: src/components/CollapsableItem.tsx [hunk 7]
```diff
@@ -38,6 +38,7 @@ const StyledCollapsableHeader = styled.div`
 
 const StyledCollapsableItem = styled.div`
     position: relative;
+    padding-bottom: 1px;
```
Comment (9teen90nine) [inline src/components/CollapsableItem.tsx:41]: 🥇

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**